### PR TITLE
!! Command

### PR DESCRIPTION
**Pull request by etp12**
## Overview

This command allows the user to append text to their previous message. The bot will resend the new message back to them. 

Example:
`User: http://google.`
`User: !!com`
`Bot: http://google.com`

If the new message is a command, the bot will also run it. This is useful if you forget a parameter of a command.

Example: 
`User: !gif`
`User: !! cats`
`Bot: !gif cats`

alt ex1

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -72,15 +72,21 @@ bot.on('disconnected', () => {
...
+      command = 'append';
+      suffix = msg.content.substring(2);
+    } else {
+      command = msg.content.toLowerCase().split(' ')[0].substring(1);
+      suffix = msg.content.substring(command.length + 2);
+    }
+    cmd = commands[command];
 
-    if (cmd) callCmd(cmd, command, bot, msg, suffix);
+    if (cmd || msg.content[1] === nconf.get('PREFIX')) callCmd(cmd, command, bot, msg, suffix);
```
Comment (dustinblackman) [inline server.js]: You shouldn't need this line. You set `command` on line 81 so how it was before should be fine.

File: lib/useful/append.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+function append(bot, msg, suffix) {
+  if (!suffix) {
+    bot.sendMessage(msg.channel, 'Usage: **`!!`** `text`');
+    return;
+  }
+
+  Promise.resolve(msg.channel.messages.getAll('author', msg.author))
+    .then(messages => {
+      messages.pop();
+      if (messages[0] === undefined) {
```
Comment (dustinblackman) [inline lib/useful/append.js]: `if (!messages[0]) {`

File: lib/useful/append.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        return;
+      }
+
+      let newCmd = messages.pop().content.trim().concat(suffix);
+      bot.sendMessage(msg.channel, newCmd);
```
Comment (dustinblackman) [inline lib/useful/append.js]: A few days ago I've added a patch where the bot can't accept commands from itself. Instead what I'd suggest is importing all the commands from `/lib/index.js` and recalling the correct command, passing `bot`, `msg`, and `suffix` that way.

File: server.js [hunk 4]
```diff
@@ -72,15 +72,21 @@ bot.on('disconnected', () => {
...
 
 function onMessage(msg) {
-  if (bot.user.username === msg.author.username) return;
-
   // Checks for PREFIX
   if (msg.content[0] === nconf.get('PREFIX')) {
-    let command = msg.content.toLowerCase().split(' ')[0].substring(1);
-    let suffix = msg.content.substring(command.length + 2);
-    let cmd = commands[command];
+    let cmd;
```
Comment (dustinblackman) [inline server.js]: You don't need to define this here. It's only used on line 87, best to put it there instead.

File: server.js [hunk 5]
```diff
@@ -72,15 +72,21 @@ bot.on('disconnected', () => {
...
+    let command;
+    let suffix;
```
Comment (dustinblackman) [inline server.js]: `let command, suffix`

**Comment by dustinblackman** [comment]
@etp12 Great work! Just a few comments and I'll be good to merge this. Thanks! :)

File: lib/useful/append.js (hunk 3 above)
Comment (etp12) [inline lib/useful/append.js]: That was my original strategy for this command, and it worked when I ran the bot. However having the line `import commands from '../../';` immediately causes `npm run mocha` to give a TypeError. I spent all of last night reading over the error message trying to figure out why and couldn't figure it out. [http://pastebin.com/HY8fpy1q](url) Any idea what's going on?

Comment (dustinblackman) [inline lib/useful/append.js]: Ah yes, I think I know the issue. I have another branch I'm going to merge in about 30 minutes that'll solve it.

**Comment by etp12** [comment]
Let me finish merging with the recent changes and I'll make another PR

**Pull request closed without merging**